AsyncDisplayKit/ASPagerFlowLayout.m

```diff
@@ -54,6 +56,18 @@ - (NSIndexPath *)_indexPathForVisiblyCenteredItem
   return nil;
 }
 
+- (BOOL)_dataSourceIsEmpty
+{
+  if ([self.collectionView numberOfSections]) {
+    if ([self.collectionView numberOfItemsInSection:0] == 0) {
```

Comment:
It might be more ideal to use a direct return with a ternary operator here, as right now the fall through condition is a bit hard to understand because the outer if-else case feels like it should be able to handle all possibilities (since it is not else if).  Would it be equivalent to use this same condition, but return ? YES : NO and removes the return at the end of the method?